src/vl-pill.js

```diff
@@ -28,78 +8,183 @@ export const VlPillElement = (SuperClass) => {
+      </span>
+    `;
+  }
+
+  static get _closablePillTemplate() {
+    return `
       <div class="vl-pill vl-pill--closable">
-          <slot></slot>
-        <button class="vl-pill__close" type="button">
+        <slot></slot>
+        <button id="close" class="vl-pill__close" type="button">
           <span class="vl-u-visually-hidden">Verwijderen</span>
         </button>
       </div>
-    `);
+    `;
   }
 
-  _closableChangedCallback(oldValue, newValue) {
-    this._shadow.lastElementChild.replaceWith((newValue != undefined ? this._getClosablePillTemplate() : this._getPillTemplate()));
+  static get _checkablePillTemplate() {
+    return `
+      <label class="vl-pill vl-pill--checkable" for="checkbox">
+        <input class="vl-pill--checkable__checkbox" type="checkbox" id="checkbox" name="checkbox" value="checked"/>
+        <span></span> <slot></slot>
+      </label>
+    `;
   }
-}
 
-export class VlLabelPill extends VlPillElement(NativeVlElement(HTMLLabelElement)) {
-  constructor() {
-    super();
+  get _checkbox() {
+    return this.shadowRoot.querySelector('#checkbox');
   }
 
-  connectedCallback() {
-    this.classList.add('vl-pill');
-    this.classList.add('vl-pill--checkable');
-    const input = this._inputElement;
-    input.classList.add("vl-pill--checkable__checkbox");
-    input.insertAdjacentHTML('afterend', this._inputStyleElement);
+  get _closeButton() {
+    return this.shadowRoot.querySelector('#close');
   }
 
-  get _inputElement() {
-    return this._element.querySelector('input');
+  __removeClosableEventListeners() {
+    const closeButton = this._closeButton;
+    if (closeButton) {
+      closeButton.removeEventListener('click', this._closeClicked);
+    }
   }
 
-  get _inputStyleElement() {
-    return `
-      <span></span>
-    `;
+  __removeCheckableEventListeners() {
+    const checkbox = this._checkbox;
+    if (checkbox) {
+      checkbox.removeEventListener('click', this._checked);
+    }
+  }
+
+  __removeEventListeners() {
+    this.__removeClosableEventListeners();
+    this.__removeCheckableEventListeners()
+  }
+
+  __addClosableEventListeners() {
+    const closeButton = this._closeButton;
+    if (closeButton) {
+      closeButton.addEventListener('click', this._closeClicked);
+    }
+  }
+
+  __addCheckableEventListeners() {
+    const checkbox = this._checkbox;
+    if (checkbox) {
+      checkbox.addEventListener('click', this._checked);
+    }
+  }
+
+  __addEventListeners() {
+    this.__addClosableEventListeners();
+    this.__addCheckableEventListeners();
   }
+
+  _redraw() {
+    this.__removeEventListeners();
+    this._shadow.lastElementChild.replaceWith(this._template(this._pillTemplate));
+    this.__addEventListeners();
+  }
+
+  _closeClicked() {
+    this.dispatchEvent(new CustomEvent(VlPill.EVENTS.close, { composed: true, bubbles: true }));
+  }
+
+  _checked() {
+    this.dispatchEvent(new CustomEvent(VlPill.EVENTS.check, {
+      composed: true,
+      bubbles: true,
```

Comment:
Reeds besproken met Sander of dit effectief nodig is om de bubbles+composed te doen. Ik zou denken dat je in dit geval als afnemer steeds op de pill kunt listenen, zonder dat je het event moet bubblen. Lijkt me vooral nodig als het event in de shadow dom van het element zelf ergens wordt gedispatched. Nog meningen?

Reply:
Kan weggelaten worden, dus ik laat ze weg :)